src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -24,18 +24,28 @@ public CommandResult execute(Model model) throws CommandException {
             if (history.isUndoneEmpty()) {
                 return new CommandResult(MESSAGE_FAILURE);
             } else {
-                System.out.println(HistoryManager.getCommands().pop());
-                System.out.println(HistoryManager.getAddressBooks().pop());
+                HistoryManager.getCommands().pop();
+                HistoryManager.getAddressBooks().pop();
                 if (history.isUndoneEmpty()) {
                     return new CommandResult(MESSAGE_FAILURE);
                 }
             }
         }
-        model.undo();
-        return new CommandResult(MESSAGE_SUCCESS);
+        Command undoneCommand = model.undo();
+        if (undoneCommand instanceof TrainingCommand) {
+            return new CommandResult(MESSAGE_SUCCESS + undoneCommand
+                + " Success!", ((TrainingCommand) undoneCommand).getDate(), model);
+        } else {
+            return new CommandResult(MESSAGE_SUCCESS + undoneCommand
```

Comment:
By editing the `toString` of the commands, perhaps this `MESSAGE_SUCCESS` can be more personalised.